src/include/cpu/rpp_cpu_filter.hpp

```diff
@@ -0,0 +1,568 @@
+/*
+MIT License
+
+Copyright (c) 2019 - 2024 Advanced Micro Devices, Inc.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*/
+
+#ifndef AMD_RPP_RPP_CPU_FILTER_HPP
+#define AMD_RPP_RPP_CPU_FILTER_HPP
```

Comment:
@r-abishek 
I analyzed the code files today for box filter, gaussian filter and sobel filter

Gaussian filter and sobel filter are having a close overlap w.r.t the helpers used
But box filter and gaussian filter are not having much overlap due to below reasons
- For U8 bitdepth
-  In case of box filter, we are doing all processing in integer itself for box filter for more efficient processing
-  In case of gaussian filter, we need to convert the inputs to float since the filter values also will be in float
So the below load functions used for box filter, cannot be used for gaussian filter
https://github.com/sampath1117/rpp/blob/sr/box_filter_host/src/include/cpu/rpp_cpu_filter.hpp#L339

- blend shuffle functions added for box filter are specifically for U8/I8 bitdepth. So this also cannot be used for gaussian filter
https://github.com/sampath1117/rpp/blob/sr/box_filter_host/src/include/cpu/rpp_cpu_filter.hpp#L95

- The only remaining functions are blend permute functions used for Float variants
https://github.com/sampath1117/rpp/blob/sr/box_filter_host/src/include/cpu/rpp_cpu_filter.hpp#L239

But this also cannot be used directly in gaussian filter
since in gaussian filter we multiply the permuted register with a filter value before doing addition
https://github.com/HazarathKumarM/rpp/blob/hk/gaussian_filter_host/src/modules/cpu/kernel/gaussian_filter.hpp#L380

We initially thought we can use the box filter algorithm as is and just multiply filter weights in gaussian filter
But after running experiments, we figured out that approach is not directly feasible in case of other kernels where we need to multiply filter values